designdocs/schemas_protocols_intents.md

```diff
@@ -0,0 +1,421 @@
+| update_delegation_group_via_governance                         | `creator_key: AccountId`<br/>`group_id: DelegationGroupid`<br/>`intent_ids: BoundedVec<IntentId, MAX_INTENTS_PER_GROUP>`                                            | Update a DelegationGroup via Governance                          |
+
+### 12. **Runtime Calls**<a id="runtime_calls"></a>
+
+The following new Custom Runtime functions are proposed:
+
+| Custom Runtime Function      | Parameters                                                                                                                        | Description                                                                                                                                                        |
+|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| resolve_intent_or_group_name | `protocol_name: ProtocolName`<br/>`descriptor_name: Option<DescriptorName>`                                                       | Resolve a name to a registered  ID or list of IDs                                                                                                                  |
+| check_delegation_group       | `group_id: DelegationGroupId`<br/>`msa_id: MessageSourceId`<br/>`provider_id: ProviderId`<br/>`block_number: Option<BlockNumber>` | Returns the Intents currently-defined DelegationGroup, mapped to a boolean indicating the current delegation status of that Intent for the given MSA and Provider. |
+
+### 13. **Storage**<a id="storage"></a>
+
+#### Schemas
+
+The storage structures for Schemas will not fundamentally change, except for internal changes to the `SchemaInfo`
+structure that will require migration:
+
+* `payload_location`  & `settings` will migrate to `IntentInfo`
+* addition of `intent_id`
+
+#### Intents
+
+The <a href="#intent_struct">IntentInfo storage</a> will be as follows:
+
+```rust
+#[pallet::storage]
+pub(super) type IntentInfos<T: Config> =
+StorageMap<_, Twox64Concat, IntentId, IntentInfo, OptionQuery>;
+```
+
+#### Delegations
+
+There will be no change to Delegation storage; existing delegated `SchemaIds` will be interpreted as `IntentIds`.
+
+#### Delegation Groups
+
+The <a href="#delegation_group_struct">DelegationGroup structures</a> will be stored as follows:
+
+```rust
+#[pallet::storage]
+pub(super) type DelegationGroups<T: Config> =
+StorageMap<_, Twox64Concat, DelegationGroupId, DelegationGroup, OptionQuery>;
+```
+
+#### Name Registry
```

Comment:
One note for discussion:

Instead of a monolithic name registry, we could define `Protocol` as a first-level storage entity with an ID and a name mapping name->ProtocolId; then define separate maps (ProtocolId, Descriptor) -> IntentId, and (ProtocolId, Descriptor) -> DelegationGroupId

Not sure what that buys us, but it's an option.

Reply:
One thing it buys is smaller, cheaper extrinsic payloads when creating/updating Intents and DelegationGroups--instead of passing both payload and descriptor names, could pass just `ProtocolId` and descriptor name.